openfeature/client_test.go

```diff
@@ -1112,3 +1113,195 @@ func TestFlagMetadataAccessors(t *testing.T) {
 		}
 	})
 }
+
+// The client MUST define a provider status accessor which indicates the readiness of the associated provider.
+// with possible values NOT_READY, READY, STALE, ERROR, or FATAL.
+func TestRequirement_1_7_1(t *testing.T) {
+
+	client := NewClient("test-client")
+
+	type requirements interface {
+		State() State
+	}
+
+	var clientI interface{} = client
+	if _, ok := clientI.(requirements); !ok {
+		t.Fatal("client des not define a status accessor which indicates the readiness of the associated provider")
+	}
+
+	TestRequirement_5_3_5(t)
+}
+
+// The client's provider status accessor MUST indicate READY if the initialize function of the associated provider
+// terminates normally.
+func TestRequirement_1_7_2(t *testing.T) {
+	defer t.Cleanup(initSingleton)
+	eventually(t, func() bool {
+		return GetApiInstance().GetClient().State() == NotReadyState
+	}, time.Second, 100*time.Millisecond, "expected client to report NOT READY state")
+
+	provider := struct {
+		FeatureProvider
+		EventHandler
+	}{
+		NoopProvider{},
+		&ProviderEventing{
+			c: make(chan Event, 1),
+		},
+	}
+
+	if err := SetProviderAndWait(provider); err != nil {
+		t.Fatalf("failed to set up provider: %v", err)
+	}
+	eventually(t, func() bool {
```

Comment:
I think we can get rid of all the `eventually`'s here, since you are using `SetProviderAndWait`.

I also think it might reduce test interference (if you see any) if you pass `t.Name()` into all `GetClient` calls as the domain: `GetApiInstance().GetClient(t.Name())`

Reply:
I think that's a good idea :) will do. And you are absolutely right, no need for `eventually` in a lot of places 👌